controllers/medusa/medusabackupschedule_controller.go

```diff
@@ -0,0 +1,140 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package medusa
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	medusav1alpha1 "github.com/k8ssandra/k8ssandra-operator/apis/medusa/v1alpha1"
+	cron "github.com/robfig/cron/v3"
+)
+
+type Clock interface {
+	Now() time.Time
+}
+
+// MedusaBackupScheduleReconciler reconciles a MedusaBackupSchedule object
+type MedusaBackupScheduleReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+	Clock  Clock
+}
+
+type RealClock struct{}
+
+func (r *RealClock) Now() time.Time {
+	return time.Now()
+}
+
+//+kubebuilder:rbac:groups=medusa.k8ssandra.io,namespace="k8ssandra",resources=medusabackupschedules,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=medusa.k8ssandra.io,namespace="k8ssandra",resources=medusabackupschedules/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=medusa.k8ssandra.io,namespace="k8ssandra",resources=medusabackupschedules/finalizers,verbs=update
+// +kubebuilder:rbac:groups=medusa.k8ssandra.io,namespace="k8ssandra",resources=cassandrabackups,verbs=get;list;watch;create
```

Comment:
We may want to wait for the remote restore work to land and generate MedusaBackupJob instances instead of CassandraBackup ones.
wdyt?

Reply:
For sure.